File: geometries.cpp
class Rectangle{
    protected:
        int width;
        int height;
    
    public:
        Rectangle(int width, int height) : width(width), height(height){}
        int get_area(){return width*height;}
        int get_width(){return height;}
};

class Square: public Rectangle{
    public:
        Square(int length) : Rectangle(length, length){}
        int get_length(){return width;}
};

File: geometries.rs
struct Rectangle{
    width: i32;
    height: i32;
}

impl Rectangle{
    pub fn new(width: i32, height: i32) -> Self{
        Self{width, height}
    }
}

struct Square{
    length: i32;
}

impl Square{
    pub fn new(length: i32) -> Self{
        Self(length)
    }
    pub fn get_length(&self) -> i32{
        self.length
    }
}